packages/typespec-ts/src/utils/clientUtils.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  SdkClient,
+  SdkContext,
+  listClients
+} from "@azure-tools/typespec-client-generator-core";
+import { Namespace } from "@typespec/compiler";
+
+export function getRLCClients(dpgContext: SdkContext): SdkClient[] {
+  const clientsInSdk = listClients(dpgContext);
+  const sdkClientsNamespacesMap = new Map<Namespace, SdkClient[]>();
+  const validClients: SdkClient[] = [];
+  clientsInSdk.forEach((client) => {
+    if (!sdkClientsNamespacesMap.has(client.service)) {
```

Comment:
since you are using `?? []` when getting the array, there's no need to do this extra `set`.